eox_nelp/openedx_filters/tests/xapi/tests_filters.py

```diff
@@ -263,41 +278,49 @@ def test_update_description(self, get_course_mock):
             returned_activity.definition.description,
         )
 
+    @data(constants.XAPI_ACTIVITY_MODULE, constants.XAPI_ACTIVITY_QUESTION)
     @patch("eox_nelp.openedx_filters.xapi.filters.get_course_from_id")
-    def test_empty_label(self, get_course_mock):
+    def test_empty_label(self, definition_type, get_course_mock):
         """ Test case when the item markdown doesn't contain a label.
 
         Expected behavior:
             - Definition description is an empty dict.
         """
         get_course_mock.return_value = self.course
+        self.set_activity(definition_type)
 
         returned_activity = self.filter.run_filter(transformer=self.transformer, result=self.activity)["result"]
 
         self.assertEqual({}, returned_activity.definition.description)
 
+    @data(constants.XAPI_ACTIVITY_MODULE, constants.XAPI_ACTIVITY_QUESTION)
     @patch("eox_nelp.openedx_filters.xapi.filters.get_course_from_id")
-    def test_default_language(self, get_course_mock):
+    def test_default_language(self, definition_type, get_course_mock):
         """ Test case when the course has no language or is not valid.
 
         Expected behavior:
             - Definition name has the default key language.
         """
         get_course_mock.return_value = {}
+        self.set_activity(definition_type)
 
         returned_activity = self.filter.run_filter(transformer=self.transformer, result=self.activity)["result"]
 
         self.assertEqual([DEFAULT_LANGUAGE], list(returned_activity.definition.name.keys()))
 
     def test_invalid_event(self):
-        """ Test case when the event name is different from edx.grades.problem.submitted.
+        """ Test case when the event definition type is different from http://adlnet.gov/expapi/activities/module
+        or http://adlnet.gov/expapi/activities/question.
 
         Expected behavior:
             - Returned activity is the same input activity.
         """
         self.transformer.event = {"name": "other-event"}
         activity = Activity(
             id="empty-activity",
+            definition=ActivityDefinition(
+                type="invalid-type",
+            )
         )
 
         returned_activity = self.filter.run_filter(transformer=self.transformer, result=activity)["result"]
```

Comment:
Like in the above comment, due the removal of this line https://github.com/eduNEXT/eox-nelp/blob/af74b95fba3c903505f78065cb873fe8240a3616/eox_nelp/openedx_filters/xapi/filters.py#L135
In the past test for eg modulestore were not called. But in this test yes, https://github.com/eduNEXT/eox-nelp/pull/121/files#diff-7f85cc618f32f4937416b9a541239a453597fac05c0552e3c34011f6faa0aaadR144

So I don't understand the invalid type if all the logic is also run. Maybe the logic doesn't change anything 🤔???🤔🤔
Or you are only filtering by the definiton and not the event-name.
https://github.com/eduNEXT/eox-nelp/blob/de1c32e0b8eba3447cc306c11e5dd65cc7812de4/eox_nelp/openedx_filters/xapi/filters.py#L138

Reply:
I think this is clearer with the other answers, if not let me know